assets/javascripts/initializers/discourse-math.js.es6

```diff
@@ -3,17 +3,28 @@ import loadScript from 'discourse/lib/load-script';
 
 let initializedMathJax = false;
 
+// The site settings are grabbed in the exported initialize function.
+// I'm not sure what the best way to make that variable available to the
+// other functions in the file. I decided to introduce a variable that's
+// local to this file but available accross the file. I'm curious if there's
+// a better approach?
+// Mark McClure, while adding a zoom on click option.
+let file_site_settings;
```

Comment:
Thanks fore the comments Sam. The links you point to do not seem to be immediately related to the initialization code; rather, they seem to be related to the markdown code here:
https://github.com/discourse/discourse-math/blob/master/assets/javascripts/lib/discourse-markdown/discourse-math.js.es6

In fact (with your information), it was very easy for me to modify that file so that `inlineMath` like 'two $x$s' parses the way I'd expect. 

On the other hand, dealing with `blockMath` has presented a greater challenge. You mentioned over on Meta that getting stuff like $$y=f(x)$$ (or even$$y=f(x)$$like, with no space) should be "trivial". If there's something I'm missing there, I'd love to know what little change could get that to work!

Honestly, though, as currently implemented, the parser has a lot of strangeness. I'd love to work on it more but, right now, my understanding of markdown-it is rather limited. For example, I see that the functions `inlineMath` and `blockMath` are responsible for detecting and formatting math, but I I don't really understand how or when those functions fire.

For the time being, if you are happy with the `let file_settings` approach to detecting the file settings (or if you can suggest a better way), I'd be happy to add the zoom_on_click and a few more settings (like the accessibility menu). Also, if you can indicate the "trivial" change to get the `blockMath` to work without spaces, I'd be glad to add that too along with the similar thing for `inlineMath`.  Of course, defaults would be `false` and I wouldn't pass the `settings` object as you recommended.